File: calc.lua
-- Calculator Lua

-- A claculator function
function calc(a, b, op)
    if op == "+" then
        return a + b
    elseif op == "-" then
        return a - b
    elseif op == "*" then
        return a * b
    elseif op == "/" then
        if b == 0 then
            return "Error: division by zero"
        else
            return a / b
        end
    else
        return "Invalid operator"
    end
end

-- repeat op until op is quit then exit
repeat
    -- ask what operator to be inputted
    io.write("Enter operator (+, -, *, /, quit): ")
    op = io.read()
    -- if op is quit then exit
    if op == "quit" then
        break
    end
    -- ask for num1
    num1 = io.read()
    -- ask for num2
    io.write("Enter second number: ")
    num2 = io.read()
    -- print result format num1 op num2 = result
    print(num1 .. " " .. op .. " " .. num2 .. " = " .. calc(num1, num2, op))
until op == "quit"

File: calc.kt
import java.util.*;


// main function
public class Main {
    public static void main(String[] args){
        try {
            while (true) {
                String input_ops = input_ops();
                int num1 = input_num();
                int num2 = input_num();
                int calculate_ops = calculate_ops(input_ops, num1, num2);
            }
        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}

// input_ops function where it returns the operator
public static String input_ops(){
    Scanner scanner = new Scanner(System.in);
    String operator = "";
    try {
        System.out.print("Enter an operator: ");
        ops = scanner.nextLine();
        if(!ops.equals("+") && 
            !ops.equals("-") && 
            !ops.equals("*") && 
            !ops.equals("/") && ){
                System.out.println("Error: Invalid operator");
                input_ops();
        }
        else if(ops.equals("q")){
            System.exit(0);
        }
    } catch (Exception e) {
        System.out.println("Error: " + e.getMessage());
    }
    return operator;
}

// input_num function where it returns the number
public static int input_num(){
    Scanner scanner = new Scanner(System.in);
    int num = 0;
    try {
        System.out.print("Enter a number: ");
        num = scanner.nextInt();
    } catch (Exception e) {
        System.out.println("Error: " + e.getMessage());
    }
    return num;
}

// calculate_ops function where it returns the result in float
public static float calculate_ops(String input_ops, int num1, int num2){
    float result = 0;
    try {
        if(input_ops.equals("+")){
            result = num1 + num2;
        }
        else if(input_ops.equals("-")){
            result = num1 - num2;
        }
        else if(input_ops.equals("*")){
            result = num1 * num2;
        }
        else if(input_ops.equals("/")){
            if(num2 == 0){
                System.out.println("Error: Cannot divide by 0");
                input_ops();
            }
            else{
                result = num1 / num2;
            }
        }
    } catch (Exception e) {
        System.out.println("Error: " + e.getMessage());
    }
    return result;
}